requirements.txt

```
@@ -1,2 +1,2 @@
-ops==1.3.0
+ops==2.0.0
```

```suggestion
ops==2.1.1
```

Any reason not to use latest ops release?

---

The main reason I upgraded to specifically ops 2.0.0 was to get around the metaclass issue I had identified in the `data_interfaces` charm library. Due to how the `DatabaseRequires` and `DatabaseProvides` classes were implemented, they inherit from both `abc.ABC` and `ops.framework.Object`. In ops 1.x, `Object` has its own custom metaclass. Python classes cannot inherit from multiple parent classes if the parent classes have conflicting metaclasses. `Object` no longer implements its own metaclass in ops 2.x, so I had to upgrade for the `data_interfaces` charm library to work.

There's no real reason to use ops 2.0.0 other than I just need ops 2.x, so I could make the requirements.txt file:

```suggestion
ops>=2.0.0
```

if you are cool with that. This will just tell pip to pull the latest version of ops 2.x.

---

Thanks for the detailed explanation. +1 to this

---

Sweet, I set the ops requirement to `ops==2.*` in https://github.com/omnivector-solutions/slurmdbd-operator/pull/7/commits/8abe4d01cdbccda2a4b3f07141304f63682676e1. This way we will get bump to a new version of ops 2.x every time a new version of ops 2.x is released, but we won't accidentally get upgraded to ops 3.x if there is a new major release.